src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -58,11 +68,13 @@ def _make_resource_with_values(
         resource: XMLResource,
         bitstream_information: BitstreamInfo | None,
     ) -> dict[str, Any]:
+        res_bnode = BNode()
+        namespaces = make_namespace_dict_from_onto_names(self.project_onto_dict)
```

Comment:
AFAIU, the namespaces remain the same for the entire lifetime of the class, so it would be more efficient to have it as class field

Reply:
The namespaces really only belong into this step of the code. I did not want to have to add another field in the class that should not actually be responsible for serialisation. Since we don't actually hand over the resource class into the functions that serialise the properties, I didn't see the advantage.